notebooks/246-paint-by-example/246-paint-by-example.ipynb

```diff
@@ -0,0 +1,943 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Paint By Example: Exemplar-based Image Editing with Diffusion Models\n",
+    "\n"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Stable Diffusion in Diffusers library\n",
+    "To work with Stable Diffusion, we will use the Hugging Face [Diffusers](https://github.com/huggingface/diffusers) library. To experiment with in-painting we can use Diffusers which exposes the [StableDiffusionInpaintPipeline](https://huggingface.co/docs/diffusers/using-diffusers/conditional_image_generation) similar to the [other Diffusers pipelines](https://huggingface.co/docs/diffusers/api/pipelines/overview). The code below demonstrates how to create `StableDiffusionInpaintPipeline` using `stable-diffusion-2-inpainting`.\n",
+    "To create the drawing tool we will install Gradio for handling user interaction.\n",
+    "\n",
+    "This is the overall flow of the application:\n",
+    "![Flow Diagram](https://user-images.githubusercontent.com/103226580/236954918-f364b227-293c-4f78-a9bf-9dcebcb1034a.png)\n",
+    "\n",
+    "This is the detailed flowchart for the pipeline:\n",
+    "![pipeline-flowchart](https://github.com/openvinotoolkit/openvino_notebooks/assets/103226580/cde2d5c4-2540-4a45-ad9c-339f7a69459d)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%pip install gradio\n",
```

Comment:
Please specify a specific version of Gradio to ensure compatibility, as certain classes and parameters used in the current Gradio interface might be deprecated in the latest version. For example, classes like `gr.Box` and parameters such as `source`, `tool`, etc., may no longer be supported in newer releases. It is recommended to update the Gradio app code to align with the latest version of Gradio and explicitly state the version used. This  will help avoid conflicts and compatibility issues in future updates.

Reply:
Done. 
Set version to 3.50.2 so that I don't have to rewrite any of the ui components for the new version as you pointed out.